### PR TITLE
Use libsodium.js if available

### PR DESCRIPTION
**Pull request by skiptirengu**
**Please describe the changes this PR makes and why it should be merged:**
Support the Emscripten version of libsodium: [libsodium.js](https://github.com/jedisct1/libsodium.js). Mainly to avoid the long compilation time of [sodium](https://www.npmjs.com/package/sodium).

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/voice/util/Secretbox.js [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
-  };
+  }),
+};
+
+for (const libName of Object.keys(libs)) {
+  try {
+    const lib = require(libName);
+    module.exports = libs[libName](lib);
+    break;
+  } catch (err) {
```
Comment (GusCaplan) [inline src/client/voice/util/Secretbox.js]: can you replace this with
```js
catch (err) {} // eslint-disable-line no-empty
```
just for consistancy

**Comment by GusCaplan** [comment]
you need to put in the readme that it would support this module, and add it as a peer dep

**Comment by skiptirengu** [comment]
Fixed, but apparently the README had CRLF instead of LF line endings. Should I revert it?

**Comment by GusCaplan** [comment]
@skiptirengu no

File: README.md [hunk 2]
```diff
@@ -1,85 +1,87 @@
...
+The preferred audio engine is node-opus, as it performs significantly better than opusscript. When both are available, discord.js will automatically choose node-opus.
+Using opusscript is only recommended for development environments where node-opus is tough to get working.
+For production bots, using node-opus should be considered a necessity, especially if they're going to be running on multiple servers.
+
+### Optional packages
+- [bufferutil](https://www.npmjs.com/package/bufferutil) to greatly speed up the WebSocket when *not* using uws (`npm install bufferutil --save`)
+- [erlpack](https://github.com/hammerandchisel/erlpack) for significantly faster WebSocket data (de)serialisation (`npm install hammerandchisel/erlpack --save`)
+- for faster voice packet encryption/decryption you can install one of the following:
+    - [sodium](https://www.npmjs.com/package/sodium) (`npm install sodium --save`)
+    - [libsodium.js](https://github.com/jedisct1/libsodium.js) (`npm install libsodium-wrappers --save`)
```
Comment (iCrawl) [inline README.md]: Should be the npmjs link instead of the github link.

**Comment by iCrawl** [comment]
Thanks for the contribution! 🎉

**Pull request merged**